pkg/repository/manifest.go

```diff
@@ -56,27 +58,98 @@ type ValidManifest interface {
 type ty struct {
 	filename  string
 	versioned bool
+	expire    time.Duration
+	threshold uint
 }
 
+// meta configs for different manifest types
 var types = map[string]ty{
-	"root":      {"root.json", true},
-	"index":     {"index.json", true},
-	"component": {"", true},
-	"snapshot":  {"snapshot.json", false},
-	"timestamp": {"timestamp.json", false},
+	"root": {
+		filename:  "root.json",
+		versioned: true,
+		expire:    time.Hour * 24 * 365, // 1y
+		threshold: 3,
+	},
+	"index": {
+		filename:  "index.json",
+		versioned: true,
+		expire:    time.Hour * 24 * 365, // 1y
+		threshold: 1,
+	},
+	"component": {
+		filename:  "",
+		versioned: true,
+		expire:    time.Hour * 24 * 365, // 1y
+		threshold: 1,
+	},
+	"snapshot": {
+		filename:  "snapshot.json",
+		versioned: false,
+		expire:    time.Hour * 24, // 1d
+		threshold: 1,
+	},
+	"timestamp": {
+		filename:  "timestamp.json",
+		versioned: false,
+		expire:    time.Hour * 24, // 1d
+		threshold: 1,
+	},
 }
 var knownVersions = map[string]struct{}{"0.1.0": {}}
 
+// Role is the metadata of a manifest
+type Role struct {
+	URL       string               `json:"url"`
+	Keys      map[string]*KeyStore `json:"keys"`
```

Comment:
This shouldn't be a map to KeyStore, it should be some new KeyInfo struct or something like that